File: c/problems/problem#2444.c
long long countSubarrays(int *nums, int numsSize, int minK, int maxK) {
  long long ans = 0;
  int lastIdxMinK = -1, lastIdxMaxK = -1;
  int startIdx = 0;
  for (int i = 0; i < numsSize; i++) {
    if (nums[i] > maxK || nums[i] < minK) {
      lastIdxMaxK = -1;
      lastIdxMinK = -1;
      startIdx = i + 1;
      continue;
    }
    if (nums[i] == minK) {
      lastIdxMinK = i;
    }
    if (nums[i] == maxK) {
      lastIdxMaxK = i;
    }
    int idx = lastIdxMaxK < lastIdxMinK ? lastIdxMaxK : lastIdxMinK;
    if (idx != -1) {
      ans += (idx - startIdx) + 1;
    }
  }
  return ans;
}

File: go/problem#2444.go
package main

func countSubarrays2444(nums []int, minK int, maxK int) int64 {
	lastIdxMinK, lastIdxMaxK := -1, -1
	startIdx := 0
	var ans int64 = 0
	for i, n := range nums {
		if n < minK || n > maxK {
			lastIdxMinK = -1
			lastIdxMaxK = -1
			startIdx = i + 1
			continue
		}
		if n == minK {
			lastIdxMinK = i
		}
		if n == maxK {
			lastIdxMaxK = i
		}
		idx := min(lastIdxMaxK, lastIdxMinK)
		if idx != -1 {
			ans += int64(idx-startIdx) + 1
		}
	}
	return ans
}
